cmd/promlts/sidecar.go

```diff
@@ -93,7 +116,56 @@ func runSidecar(
 			s.Stop()
 			l.Close()
 		})
+	}
+	// Periodically query the Prometheus config. We use this as a heartbeat as well as for updating
+	// the external labels we apply.
+	{
+		promUp := prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "promlts_sidecar_prometheus_up",
+			Help: "Boolean indicator whether the sidecar can reach its Prometheus peer.",
+		})
+		lastHeartbeat := prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "promlts_sidecar_last_heartbeat_seconds",
+			Help: "Second timestamp of the last attempted heartbeat",
+		})
+		reg.MustRegister(promUp, lastHeartbeat)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		tick := time.NewTicker(30 * time.Second)
+
+		do := make(chan struct{}, 1)
+		do <- struct{}{}
+
+		go func() {
+			for range tick.C {
+				do <- struct{}{}
+			}
+		}()
 
+		g.Add(func() error {
+			for {
+				select {
+				case <-ctx.Done():
+					return nil
+				case <-do:
+					elset, err := queryExternalLabels(ctx, baseURL)
+					if err != nil {
+						level.Warn(logger).Log("msg", "heartbeat failed", "err", err)
+						promUp.Set(0)
+					} else {
+						promUp.Set(1)
+
+						mtx.Lock()
+						externalLabels = elset
+						mtx.Unlock()
+					}
+					lastHeartbeat.Set(float64(time.Now().Unix()))
```

Comment:
well... it hearthbeat failed, do we want to set this gauge? Or counter if we decide on that (see 2 comments above)?

Reply:
or you want to just cover "attempts" then we need to name it accordingly?

Reply:
Ah, you already said it here as well. Definitely better to just set it on success.